crates/nu-parser/src/parse_keywords.rs

```diff
@@ -1005,7 +1005,27 @@ pub fn parse_alias(
             working_set.add_decl(Box::new(decl));
         }
 
-        if spans.len() < 4 {
+        // special case for `alias foo=bar`
+        if spans.len() == 2 && working_set.get_span_contents(spans[1]).contains(&b'=') {
+            // convert second argument to string.  Its a bit messy, could use
+            // refactoring.
+            let arg = working_set.get_span_contents(spans[1]).to_owned();
+            let arg = String::from_utf8_lossy(&arg);
+            // invariant: `arg` must contain the '=' byte from the if check
+            let index = arg.rfind('=').unwrap();
```

Comment:
No unwraps, these need to be handled manually.

Reply:
Also, you can probably use just `.find()` instead of `.rfind()` and avoid allocating a temporary Vec<u8> and String.

Reply:
How should I handle the lack of `=`?  This technically should never happen because the `if` condition checks that `=` is a part of the byte slice.  And I believe even multi-byte UTF8 characters can't have it in latter bytes.

A malformed string would cause a panic, though.

Reply:
I replaced it with `.unwrap_or(arg.len())`, which should do the same thing, but without implicit possibility of panicking.